protocol/src/main/java/org/fido/iot/protocol/CryptoService.java

```diff
@@ -579,6 +581,113 @@ public void verify(Composite chain) {
     }
   }
 
+  /**
+   * Verifies certificate algorithm.
+   *
+   * @param algorithmName Public key algorithm
+   */
+  public void verifyAlgorithm(String algorithmName) throws InvalidOwnershipVoucherException {
+    if (!(algorithmName.equals(Const.EC_ALG_NAME))) {
+      throw new InvalidOwnershipVoucherException(
+          "Wrong public key algorithm inside the device certificate - supported: ECDSA, received: "
+              + algorithmName);
+    }
+  }
+
+  /**
+   * Verifies certificate public key curve.
+   *
+   * @param ecdsaCurveName ECDSA curve name
+   */
+  private void verifyCurveName(String ecdsaCurveName) throws InvalidOwnershipVoucherException {
+    if (!(ecdsaCurveName.contains(Const.SECP256R1_CURVE_NAME)
+        || ecdsaCurveName.contains(Const.SECP384R1_CURVE_NAME))) {
+      throw new InvalidOwnershipVoucherException("Mismatch of ECDSA curve type.");
+    }
+  }
+
+  /**
+   * Verifies certificate public key size.
+   *
+   * @param pubKeySize Public key size
+   */
+  private void verifyKeySize(int pubKeySize) throws InvalidOwnershipVoucherException {
+    if (pubKeySize != Const.BIT_LEN_256 && pubKeySize != Const.BIT_LEN_384) {
```

Comment:
should be or condition?
`pubKeySize != Const.BIT_LEN_256 || pubKeySize != Const.BIT_LEN_384`?

Reply:
The proposed `||` condition in the if statement would throw exception even if one of the condition is not satisfied. Let's say key size is 256 bits in that case the second condition will fail and throw exception and when key size is 384 bits, the first condition will fail and throw exception. However the purpose behind this check is to throw exception when key size is neither 256 bits or 384 bits, so `&&` condition is appropriate.

Reply:
Thanks for the explanation!
I misread not equal to condition.

Reply:
No problem!!